### DISCUSSION TITLE
how to interpret (and fix) this situation?

### DISCUSSION THREAD

**Discussion by rpodgorny** [moosefs/moosefs#554] [General]
hi,

after a "double disk failure situation" (just for a second but still) i managed to fully recorver all chunks and therefore all files. at least i hope so - when looking at the gui table of chunks, one chunk still keeps showing up in the "zero out of two (two is the goal)" copies state (thus "missing" or "invalid"?) but! i see no invalid files being listed in the filesystem check section neither i see any errors in log on master and chunkservers. it's been like this for over a week. :-(

so my question is: how do i locate the "missing" chunk? is is actually missing? may it be just some inner inconsistency in master? how to further debug this?

thanks a lot for any advice!

**Comment by chogata** [comment]
It takes MooseFS one full file loop to find missing files. How long is your file loop and was at least one completed after you recovered from the failure? You will find the file loop information on CGI "Info" tab (under "Filesystem check info"). If you had one full file loop and no missing files, perhaps this is a "locked, unused" chunk (check under "Chunk operations info"). In that case this is something that you don't have to worry about and it will disappear in about 1 week.
If you are not sure, you can post a screenshot of your info tab here.

**Comment by rpodgorny** [comment]
![screenshot](https://github.com/moosefs/moosefs/assets/2897520/a09f8ec6-a0e0-48fc-bc96-145a83a30298)

**Comment by rpodgorny** [comment]
wow, thanks for such quick reply!

hmm, so it seems it's the "locked, unused" case... (but why is there "4", anyway?) ...so i just wait some more...

**Comment by chogata** [comment]
Locked unused chunks are chunks that are not connected to any file. They are a result of an operation interrupted due to some hardware failure (in your case whatever caused your "double disk failure"). 99.99% of the time they are completely redundant, whether they even physically exist or not (in your case 3 exist, 1 doesn't). But once in a blue moon a weird coincidence* might result in a chunk with some important data, that could be manually recovered. So we do not delete those unconnected chunks for a while - just in case. We do a lot of things just in case, but that's what makes MooseFS more safe for your data :)

*the only scenario, that has some probability of creating unconnected chunks that are not redundant (aka contain some valuable data), would be a power failure of several MooseFS components, including the master, at the same time, and the data affected would be the data written during(just before) the failure; still, we are cautious, always

**Comment by rpodgorny** [comment]
ok, thanks a lot for the explanation. so if there's no way to force the unlock, i'll just wait a little longer and wait for the missing chunk to disappear by itself...

also, let me thank you for your exceptional support! i really do appreciate the effort - even more so when you do it even for non-paying "customers"...!
